File: solutions/1894. Find the Student that Will Replace the Chalk/1894.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def chalkReplacer(self, chalk: List[int], k: int) -> int:
    k %= sum(chalk)

    for i, c in enumerate(chalk):
      if k < c:
        return i
      k-= c

File: solutions/1894. Find the Student that Will Replace the Chalk/1894.rs
// Time complexity: O(n)
// Space complexity: O(1)
impl Solution {
    pub fn chalk_replacer(chalk: Vec<i32>, k: i32) -> i32 {
        let total_chalk: i64 = chalk.iter().map(|&x| x as i64).sum();
        let mut k = k as i64 % total_chalk;

        for (i, &c) in chalk.iter().enumerate() {
            if k < c as i64 {
                return i as i32;
            }
            k -= c as i64;
        }

        0
    }
}
